frontend/src/pages/GamePage.tsx

```diff
@@ -1,99 +1,200 @@
+
+      if (response.ok) {
+        return true; // Successfully submitted answer
+      } else {
+        throw new Error(result.message || "Failed to submit answer.");
+      }
+    } catch (error) {
+      console.error("Error submitting answer:", error);
+      setTerminalLineData((prevData) => [
+        ...prevData,
+        `Error: ${error instanceof Error ? error.message : "Unknown error"}`,
+      ]);
+      return false;
+    }
+  };
+
+  const handleInput = async (input: string) => {
+    if (gameOver) {
+      setTerminalLineData((prevData) => [
+        ...prevData,
+        "Game is over. No more input allowed.",
+      ]);
+      return;
+    }
+
+    const currentQuestion = questions[currentQuestionIndex];
+    const selectedOptionIndex = parseInt(input) - 1;
+
+    if (
+      isNaN(selectedOptionIndex) ||
+      selectedOptionIndex < 0 ||
+      selectedOptionIndex >= currentQuestion.options.length
+    ) {
+      setTerminalLineData((prevData) => [
+        ...prevData,
+        "Invalid input. Please choose a valid option.",
+      ]);
+      return;
+    }
+
+    const selectedOption = currentQuestion.options[selectedOptionIndex];
+    const newStats = {
+      health: stats.health + selectedOption.points.health,
+      wealth: stats.wealth + selectedOption.points.wealth,
+      happiness: stats.happiness + selectedOption.points.happiness,
+    };
+
+    // Immediate frontend update
+    setTerminalLineData((prevData) => [
+      ...prevData,
+      `You chose: ${selectedOption.text}`,
+      `Updated stats - Health: ${newStats.health}, Wealth: ${newStats.wealth}, Happiness: ${newStats.happiness}.\n`,
+    ]);
+
+    // Sync stats with database
+    const isAnswerSubmitted = await submitAnswer(
+      currentQuestion._id,
+      selectedOptionIndex.toString(),
+      newStats
     );
+
+    if (isAnswerSubmitted) {
+      // Update React state with synced stats
+      setStats(newStats);
+
+      // Move to the next question or end the game
+      const nextQuestionIndex = currentQuestionIndex + 1;
+      if (nextQuestionIndex < questions.length) {
+        setCurrentQuestionIndex(nextQuestionIndex);
+
+        const nextQuestion = questions[nextQuestionIndex];
+        const nextQuestionText = `Question ${nextQuestionIndex + 1}: ${nextQuestion.title}`;
+        const nextOptionsText = nextQuestion.options
+          .map((option: Option, index: number) => `${index + 1}. ${option.text}`)
+          .join("\n");
+        setTerminalLineData((prevData) => [
+          ...prevData,
+          nextQuestionText,
+          nextOptionsText,
+        ]);
+      } else {
+        setGameOver(true);
+        setTerminalLineData((prevData) => [
+          ...prevData,
+          `Game Over! Final stats - Health: ${newStats.health}, Wealth: ${newStats.wealth}, Happiness: ${newStats.happiness}.\nThank you for playing!`,
+        ]);
+      }
+    }
+  };
+
+  if (isLoading) {
+    return <div>Loading questions...</div>;
+  }
```

Comment:
Remove the loading from here, it should be displayed inside the Terminal component. The reason is, currently it will just render loading on the page but it should be rendered on the Terminal itself that the questions are currently loading, we don't want to change that ui.